packages/strapi/lib/Strapi.js

```diff
@@ -414,35 +416,48 @@ class Strapi {
     return reload;
   }
 
-  async runBootstrapFunctions() {
-    const execBootstrap = async fn => {
+  async runLifecyclesFunctions(functionName) {
+    const execFunction = async fn => {
       if (!fn) return;
 
       return fn();
     };
 
-    // plugins bootstrap
+    const configPath = `functions.${functionName}`;
+
+    // plugins register
     const pluginBoostraps = Object.keys(this.plugins).map(plugin => {
-      return execBootstrap(_.get(this.plugins[plugin], 'config.functions.bootstrap')).catch(err => {
-        strapi.log.error(`Bootstrap function in plugin "${plugin}" failed`);
+      const pluginFunc = _.get(this.plugins[plugin], `config.${configPath}`);
+
+      return execFunction(pluginFunc).catch(err => {
+        strapi.log.error(`${functionName} function in plugin "${plugin}" failed`);
         strapi.log.error(err);
         strapi.stop();
       });
     });
+
     await Promise.all(pluginBoostraps);
 
-    // user bootstrap
-    await execBootstrap(_.get(this.config, ['functions', 'bootstrap']));
+    // user register
+    await execFunction(_.get(this.config, configPath));
 
-    // admin bootstrap : should always run after the others
-    const adminBootstrap = _.get(this.admin.config, 'functions.bootstrap');
-    return execBootstrap(adminBootstrap).catch(err => {
-      strapi.log.error(`Bootstrap function in admin failed`);
+    // admin register : should always run after the others
+    const adminFunc = _.get(this.admin.config, configPath);
+    return execFunction(adminFunc).catch(err => {
+      strapi.log.error(`${functionName} function in admin failed`);
       strapi.log.error(err);
       strapi.stop();
     });
   }
 
+  async runBootstrapFunctions() {
+    return this.runLifecyclesFunctions('bootstrap');
+  }
+
+  async runRegisterFunctions() {
+    return this.runLifecyclesFunctions('register');
+  }
```

Comment:
What is the purpose of splitting the bootstrap files in 2 (bootstrap & register)? Is it to run all the registrations after the bootstrap? If yes, what usecase do you have in mind?

Reply:
we need the register to change the content types before the db is initialized. you register then bootstrap. This will allow even more things in the futur

Reply:
Should line 453 and 457 be inversed then ?
Happy to talk about it IRL

Reply:
Will remove those two functions actually